openfecli/commands/gather.py

```diff
@@ -0,0 +1,133 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+
+import click
+from openfecli import OFECommandPlugin
+import pathlib
+from openfecli.utils import write
+
+
+def is_results_json(f):
+    # sanity check on files before we try and deserialize
+    return 'estimate' in open(f, 'r').read(20)
+
+
+def load_results(f):
+    # path to deserialized results
+    import json
+    from gufe.tokenization import JSON_HANDLER
+
+    return json.load(open(f, 'r'), cls=JSON_HANDLER.decoder)
+
+
+def get_names(result):
+    # Result to list of ligand names
+    return tuple(list(result['unit_results'].values())[0]['name'].split()[:2])
+
+
+def get_type(f):
+    if 'solvent' in f:
+        return 'solvent'
+    elif 'vacuum' in f:
+        return 'vacuum'
+    else:
+        return 'complex'
+
+@click.command(
+    'gather',
+    short_help="Gather DAG result jsons for network of RFE results into single TSV file"
+)
+@click.argument('rootdir',
+                type=click.Path(dir_okay=True, file_okay=False,
+                                         path_type=pathlib.Path),
+                required=True)
+@click.argument('output',
+                type=click.File(mode='w'),
+                required=True)
+def gather(rootdir, output):
+    """Gather DAG result jsons of relative calculations and write to single tsv file
+
+    Will walk ROOTDIR recursively and find all results files ending in .json (i.e those produced by the quickrun
+    command).  Each of these is a single leg (e.g. L1-L2 in solvent) of a free energy calculation.
```

Comment:
> (e.g. L1-L2 in solvent)

I'm not sure this make sense, but also it might not be worth having here at all. I'd just say "each of these is a separate leg along a relative free energy thermodynamic cycle".